list.go

```diff
@@ -0,0 +1,91 @@
+package pack
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+)
+
+type List struct {
+	T     Type
+	Elems []Value
+}
+
+func NewList(vs ...interface{}) List {
+	elems := make([]Value, len(vs))
+	for i := range elems {
+		elems[i] = vs[i].(Value)
+	}
+	return List{
+		T:     elems[0].Type(),
```

Comment:
This will panic if the elements are empty. Consider changing the profile of `NewList` to be `(t Type, elems ...interface{})`.